content/blog/elm-review-v2/index.md

```diff
@@ -0,0 +1,269 @@
+
+## Sorry but... what is elm-review?
+
+If you missed the [initial announcement](/announcing-elm-review/) or if your memory is unclear, let me explain.
+
+`elm-review` is a static code analysis tool for Elm, which looks at your Elm project, and reports patterns that you configured it to find using "rules" in a friendly Elm compiler-like way.
+
+It is highly customizable because you can write your own rules. Since these are written in Elm, you don't need to know a different language to do so, and you can even publish them as Elm packages!
+
+I put a lot of effort into making a [**very nice API**](https://package.elm-lang.org/packages/jfmengels/elm-review/2.0.0/Review-Rule) for you to use ([here](https://github.com/jfmengels/review-simplification/blob/master/src/NoBooleanCaseOf.elm) [are](https://github.com/jfmengels/review-debug/blob/master/src/NoDebug/Log.elm) [some](https://package.elm-lang.org/packages/jfmengels/elm-review/2.0.0/Review-Rule#withSimpleExpressionVisitor) [examples](https://package.elm-lang.org/packages/jfmengels/elm-review/2.0.0/Review-Rule#withSimpleImportVisitor)), even going through the lengths of discovering new Elm techniques (expect to see blog posts on the **phantom builder pattern** on this blog), and I am very happy with the result!
+
+To some, `elm-review` looks like a linter. Which isn't necessarily wrong, since you can enable rules that help improve the quality of your code. If you dive a bit further, you will find out that it can enforce coding conventions for your team, and create new guarantees that the Elm compiler can not give you.
+
+For instance you can write rules that
+
+- report when you [pass values outside of 0-255 to `Css.rgb255`](https://package.elm-lang.org/packages/folq/review-rgb-ranges/latest/)
+- give an error if any `Regex.fromString` calls have invalid regexes (for literal strings)
+
+Unfortunately, as soon as a problem spawns multiple modules, you wouldn't be able to enforce those guarantees. But that was the case until we got...
+
+## Project rules!
+
+In `elm-review` version 1, `Elm Analyse`, `ESLint` and a lot of similar tools for other languages, the analysis is scoped to a single file.
+
+This means that when a rule has finished looking at module A and starts looking at module B, it forgets everything about module A. That means we can't answer a lot of questions that we would like the answer to, such as:
+
+- Which module does function `xyz` come from? If a module is imported using `import A exposing (..)`, then we potentially lose the ability to tell. We have the same problem with types when encountering `import A exposing (B(..))`.
+- What is the type of an imported function?
+- Is this element ever used in the project?
+
+In addition to the **module rules** that we just described, `elm-review` now has **project rules**. These rules go through all the modules and can use information collected from a different module to infer or report things, solving the problems mentioned above.
```

Comment:
```suggestion
In addition to the old **module rules**, `elm-review` now has **project rules**. These rules go through all the modules and can use information collected from a different module to infer or report things, solving the problems mentioned above.
```

Reply:
I'll keep the current wording I think, because otherwise the notion of **modules rules** pops out of nowhere. The current wording explain that module rules are the thing we just talked about.